integration/querydsl/expressions/src/main/java/com/blazebit/persistence/querydsl/JPQLNextQueryFactory.java

```diff
@@ -16,122 +16,276 @@
 
 package com.blazebit.persistence.querydsl;
 
-import com.blazebit.persistence.CriteriaBuilderFactory;
 import com.querydsl.core.Tuple;
-import com.querydsl.core.dml.DeleteClause;
 import com.querydsl.core.types.EntityPath;
 import com.querydsl.core.types.Expression;
-import com.querydsl.core.types.dsl.Expressions;
+import com.querydsl.core.types.Path;
+import com.querydsl.core.types.SubQueryExpression;
 import com.querydsl.jpa.JPQLQueryFactory;
-import com.querydsl.jpa.impl.JPADeleteClause;
-import com.querydsl.jpa.impl.JPAInsertClause;
-import com.querydsl.jpa.impl.JPAUpdateClause;
 
-import javax.persistence.EntityManager;
+import java.util.Collection;
+import java.util.List;
 
 /**
  * Query factory to simplify {@link BlazeJPAQuery} instantiation.
  *
  * @author Jan-Willem Gmelig Meyling
  * @since 1.6.2
  */
-public class JPQLNextQueryFactory implements JPQLQueryFactory {
```

Comment:
The change from class to interface is incompatible and clients must recompile their code. Is this change really necessary? Maybe it's worth adding a new subclass and promote the use of that, but still keep this class around?

Reply:
I doubt any users will currently use this class:
- Its was only recently introduced
- The documentation does not mention it or how to use it, all examples just use `new BlazeJPAQuery<>()` to construct new queries
- The class was pretty useless as it were without the signature changes.

Reply:
Ok, that's convincing!